### PR TITLE
Compress packages - only build libraries 

### PR DESCRIPTION
**Pull request by heitbaum**
- xz: only build libraries
- libarchive: only build libraries
- lz4: only build libraries

### CODE REVIEW

File: packages/compress/libarchive/package.mk [hunk 1]
```diff
@@ -12,17 +12,35 @@ PKG_DEPENDS_TARGET="cmake:host gcc:host"
 PKG_SHORTDESC="A multi-format archive and compression library."
 
 PKG_CMAKE_OPTS_TARGET="-DCMAKE_POSITION_INDEPENDENT_CODE=1 \
```
Comment (CvH) [inline packages/compress/libarchive/package.mk:14]: options sorted by a-z
```
PKG_CMAKE_OPTS_TARGET="-DCMAKE_POSITION_INDEPENDENT_CODE=1 \
                       -DBUILD_SHARED_LIBS=OFF \
                       -DENABLE_ACL=ON \
                       -DENABLE_BZip2=ON \
                       -DENABLE_CAT_SHARED=FALSE \
                       -DENABLE_CAT=OFF \
                       -DENABLE_CNG=ON \
                       -DENABLE_COVERAGE=FALSE \
                       -DENABLE_CPIO_SHARED=FALSE \
                       -DENABLE_CPIO=OFF \
                       -DENABLE_EXPAT=OFF \
                       -DENABLE_ICONV=OFF \
                       -DENABLE_INSTALL=ON \
                       -DENABLE_LIBB2=OFF \
                       -DENABLE_LIBGCC=ON \
                       -DENABLE_LIBXML2=OFF \
                       -DENABLE_LZ4=ON \
                       -DENABLE_LZMA=OFF \
                       -DENABLE_LZO=ON \
                       -DENABLE_MBEDTLS=OFF \
                       -DENABLE_NETTLE=OFF \
                       -DENABLE_OPENSSL=ON \
                       -DENABLE_PCRE2POSIX=ON \
                       -DENABLE_PCREPOSIX=OFF \
                       -DENABLE_TAR_SHARED=FALSE \
                       -DENABLE_TAR=OFF \
                       -DENABLE_TEST=OFF \
                       -DENABLE_UNZIP_SHARED=FALSE \
                       -DENABLE_UNZIP=OFF \
                       -DENABLE_WERROR=0 \
                       -DENABLE_XATTR=ON \
                       -DENABLE_ZLIB=ON \
                       -DENABLE_ZSTD=ON"
```

File: packages/compress/xz/package.mk [hunk 2]
```diff
@@ -24,9 +24,13 @@ PKG_CONFIGURE_OPTS_HOST="--disable-shared --enable-static \
                          --disable-scripts \
                          --enable-symbol-versions=no"
 
-PKG_CONFIGURE_OPTS_TARGET="--disable-shared --enable-static \
+PKG_CONFIGURE_OPTS_TARGET="--disable-shared \
```
Comment (CvH) [inline packages/compress/xz/package.mk:27]: sorted and kept static/shared on top
```
PKG_CONFIGURE_OPTS_TARGET="--enable-static \
                           --disable-shared \
                           --disable-doc \
                           --disable-lzma-links \
                           --disable-lzmadec \
                           --disable-lzmainfo \
                           --disable-scripts \
                           --disable-xz \
                           --disable-xzdec \
                           --enable-symbol-versions=no"
```

File: packages/compress/libarchive/package.mk (hunk 1 above)
Comment (heitbaum) [reply packages/compress/libarchive/package.mk:14]: Updated sort order

File: packages/compress/xz/package.mk (hunk 2 above)
Comment (heitbaum) [reply packages/compress/xz/package.mk]: Updated sort order

**Review by CvH — APPROVED** [review]

**Pull request merged**